lib/streamlit/elements/vega_charts.py

```diff
@@ -1139,6 +1148,22 @@ def bar_chart(
 
         """
 
+        # Offset encodings (used for non-stacked/grouped bar charts) are not supported in Altair < 5.0.0
+        if type_util.is_altair_version_less_than("5.0.0") and stack is False:
+            raise StreamlitAPIException(
+                "Streamlit does not support non-stacked (grouped) bar charts with Altair 4.x. Please upgrade to Version 5. "
+                "If you would like to use Altair 4.x with this feature, please upvote "
+                "this [Github issue](https://github.com/streamlit/streamlit/issues/8954)."
```

Comment:
I might not encourage upvoting (I would not want to prioritize supporting legacy systems because a campaign to upvote is stronger than a campaign to upgrade to 5.0), but I'm not going to block on this.

Reply:
Fair point, I had just followed the pattern from selections on charts... Think I will remove